main.tf

```diff
@@ -0,0 +1,666 @@
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRoleAttachment resource attaches the CloudGuardAWPCrossAccountRolePolicy to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_role_policy_attachment" "CloudGuardAWPCrossAccountRoleAttachment" {
+  role       = aws_iam_role.CloudGuardAWPCrossAccountRole.name
+  policy_arn = aws_iam_policy.CloudGuardAWP.arn
+}
+# end resources for CloudGuardAWPCrossAccountRole
+
+# Cross account role policy
+# The CloudGuardAWPCrossAccountRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy" "CloudGuardAWPCrossAccountRolePolicy_InAccount" {
+  count       = local.scan_mode == "inAccount" ? 1 : 0
+  name        = "CloudGuardAWPCrossAccountRolePolicy_InAccount"
+  description = "Policy for CloudGuard AWP Cross Account Role"
+  tags = local.common_tags
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:CreateSecurityGroup",
+          "ec2:DescribeManagedPrefixLists",
+          "ec2:DescribeSecurityGroups",
+          "ec2:DescribeSecurityGroupRules",
+          "ec2:RevokeSecurityGroupEgress",
+          "ec2:AuthorizeSecurityGroupEgress",
+          "ec2:CreateTags",
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:DeleteSecurityGroup",
+        ]
+        Resource = "*"
+        Condition = {
+          StringEquals = {
+            "aws:ResourceTag/Owner" = "CG.AWP"
+          }
+        }
+      },
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRolePolicy_SaaS resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPCrossAccountRole in SaaS mode.
+resource "aws_iam_policy" "CloudGuardAWPCrossAccountRolePolicy_SaaS" {
+  count       = local.scan_mode == "saas" ? 1 : 0
+  name        = "CloudGuardAWPCrossAccountRolePolicy_SaaS"
+  description = "Policy for CloudGuard AWP Cross Account Role - SaaS Mode"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:DescribeKey",
+          "kms:ReplicateKey",
+        ]
+        Resource = [aws_kms_key.CloudGuardAWPKey[count.index].arn]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:PutKeyPolicy",
+          "kms:ScheduleKeyDeletion",
+          "kms:CancelKeyDeletion",
+          "kms:TagResource",
+        ]
+        Resource = aws_kms_key.CloudGuardAWPKey[count.index].arn
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "kms:CreateKey",
+        ]
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+# The CloudGuardAWPCrossAccountRolePolicyAttachment resource attaches the CloudGuardAWPCrossAccountRolePolicy to the CloudGuardAWPCrossAccountRole.
+resource "aws_iam_policy_attachment" "CloudGuardAWPCrossAccountRolePolicyAttachment" {
```

Comment:
change resource name from "CloudGuardAWPCrossAccountRolePolicyAttachment" to "CloudGuardAWPCrossAccountRolePolicyAttachment_InAccount"